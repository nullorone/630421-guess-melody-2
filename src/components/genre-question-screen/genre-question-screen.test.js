import React from 'react';
import renderer from 'react-test-renderer';
import GenreQuestionScreen from "./genre-question-screen";

const Value = {
  EMPTY: 0,
  FULL: 1
};

const EMPTY_ANSWER = {
  src: ``,
  genre: ``,
};

describe(`Render Component`, () => {
  const trackButtonClickHandler = jest.fn();
  const submitClickHandler = jest.fn();
  const initialProps = {
    time: Value.EMPTY,
    mistakes: Value.EMPTY,
    question: ``,
    answers: [EMPTY_ANSWER],
    onTrackButtonClick: trackButtonClickHandler,
    onSubmitClick: submitClickHandler,
  };

  it(`Render`, () => {
    const tree = renderer
      .create(
          <GenreQuestionScreen
            {...initialProps}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
