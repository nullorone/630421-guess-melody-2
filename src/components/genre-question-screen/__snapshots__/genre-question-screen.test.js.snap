// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Component Render 1`] = `
<section
  className="game game--genre"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <svg
      className="timer"
      viewBox="0 0 780 780"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        className="timer__line"
        cx="390"
        cy="390"
        r="370"
      />
    </svg>
    <div
      className="timer__value"
      xmlns="http://www.w3.org/1999/xhtml"
    >
      <span
        className="timer__mins"
      >
        00
      </span>
      <span
        className="timer__dots"
      >
        :
      </span>
      <span
        className="timer__secs"
      >
        00
      </span>
    </div>
    <div
      className="game__mistakes"
    />
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      
    </h2>
    <form
      className="game__tracks"
      onSubmit={[Function]}
    >
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          onClick={[MockFunction]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio
            src=""
          />
        </div>
        <div
          className="game__answer"
        >
          <input
            className="game__input visually-hidden"
            id="answer-1"
            name="answer"
            type="checkbox"
            value=""
          />
          <label
            className="game__check"
            htmlFor="answer-1"
          >
            Отметить
          </label>
        </div>
      </div>
      <button
        className="game__submit button"
        type="submit"
      >
        Ответить
      </button>
    </form>
  </section>
</section>
`;
